datadreamer/prompt_generation/synonym_generator.py

```diff
@@ -33,25 +37,41 @@ def __init__(
         self.synonyms_number = synonyms_number
         self.seed = seed
         self.device = device
-        self.model, self.tokenizer = self._init_lang_model()
+        self.model, self.tokenizer, self.pipeline = self._init_lang_model()
 
     def _init_lang_model(self):
-        """Initializes the language model and tokenizer for synonym generation."""
-        print("Initializing language model for synonym generation")
+        """Initializes the language model and tokenizer for prompt generation.
+
+        Returns:
+            tuple: The initialized language model and tokenizer.
+        """
         if self.device == "cpu":
+            print("Loading language model on CPU...")
             model = AutoModelForCausalLM.from_pretrained(
                 "mistralai/Mistral-7B-Instruct-v0.1",
                 torch_dtype="auto",
                 device_map="cpu",
                 low_cpu_mem_usage=True,
             )
         else:
+            print("Loading FP16 language model on GPU...")
             model = AutoModelForCausalLM.from_pretrained(
-                "mistralai/Mistral-7B-Instruct-v0.1", torch_dtype=torch.float16
+                "mistralai/Mistral-7B-Instruct-v0.1",
+                torch_dtype=torch.float16,
+                trust_remote_code=True,
+                device_map=self.device,
             )
 
         tokenizer = AutoTokenizer.from_pretrained("mistralai/Mistral-7B-Instruct-v0.1")
-        return model.to(self.device), tokenizer
+        pipe = pipeline(
+            "text-generation",
+            model=model,
+            tokenizer=tokenizer,
+            torch_dtype=torch.float16 if self.device == "cuda" else "auto",
+            device_map=self.device,
+        )
+        print("Done!")
+        return model, tokenizer, pipe
```

Comment:
@HonzaCuhel what is the reason for using a pipeline here?

Reply:
Since I changed the code to use the pipeline in all the other LM classes, I thought it should be consistent everywhere, but I can change it, it's no problem.

Reply:
Ok, makes sense